Libraries/Image/RCTLocalAssetImageLoader.m

```diff
@@ -50,7 +51,49 @@ - (RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL
     }
 
     NSString *imageName = RCTBundlePathForURL(imageURL);
-    UIImage *image = [UIImage imageNamed:imageName];
+    NSBundle *targetBundle = [NSBundle mainBundle];
+    
+    if (bundlePath) {
+      NSString *targetBundleName = RCTBundlePathForURL([NSURL URLWithString:bundlePath]);
+      NSString *targetBundleNameWithExtension = [NSString stringWithFormat:@"%@%@", targetBundleName, @".bundle"];
+      NSMutableArray *bundles = [[NSBundle allBundles] mutableCopy];
+      
+      // Get any bundle files included in the mainBundle and add them to the array
+      for (NSString *path in [[NSBundle mainBundle] pathsForResourcesOfType:@"bundle" inDirectory:nil]) {
```

Comment:
I don't know much about iOS development, but it seems like we're iterating through every bundle and every framework every time `loadImageForURL` is called.  Would it be possible to cache this work and only do it the first time?  Or is it possible that the bundles and frameworks can change at runtime?